tests/functional/catalogs/test_catalogs_parsing.py

```diff
@@ -0,0 +1,143 @@
+from unittest import mock
+
+import pytest
+
+from dbt.adapters.catalogs import (
+    CatalogIntegration,
+    CatalogIntegrationConfig,
+    DbtCatalogIntegrationNotFoundError,
+)
+from dbt.tests.util import run_dbt, write_config_file
+from dbt_common.exceptions import DbtValidationError
+
+
+class TestCatalogIntegration(CatalogIntegration):
+    def __init__(self, config: CatalogIntegrationConfig):
+        super().__init__(config)
+        for key, value in config.adapter_properties.items():
+            setattr(self, key, value)
+
+
+def mock_integrations():
+    """Return mock dict for catalog integrations"""
+    return {"glue": TestCatalogIntegration, "iceberg": TestCatalogIntegration}
+
+
+# Common write integration configs for reuse
+FIRST_INTEGRATION = {
+    "name": "my_write_integration",
+    "external_volume": "my_external_volume",
+    "table_format": "iceberg",
+    "catalog_type": "glue",
+    "adapter_properties": {"my_custom_property": "some_value"},
+}
+
+SECOND_INTEGRATION = {
+    "name": "my_second_write_integration",
+    "external_volume": "my_other_external_volume",
+    "table_format": "iceberg",
+    "catalog_type": "glue",
+    "adapter_properties": {"my_custom_property": "some_other_value"},
+}
+
+
+class BaseCatalogTest:
+    @pytest.fixture
+    def integrations(self):
+        """Override in subclasses to provide write_integrations"""
+        return []
+
+    @pytest.fixture
+    def active_integration(self):
+        """Override in subclasses to provide active_write_integration"""
+        return None
+
+    @pytest.fixture
+    def catalogs(self, integrations, active_integration):
+        catalog = {"name": "test_catalog", "write_integrations": integrations}
+        if active_integration:
+            catalog["active_write_integration"] = active_integration
+        return {"catalogs": [catalog]}
+
+    def setup_test(self, catalogs, project):
+        write_config_file(catalogs, project.project_root, "catalogs.yml")
+        return mock.patch.dict(project.adapter.CATALOG_INTEGRATIONS, mock_integrations())
+
+
+class TestSingleWriteIntegration(BaseCatalogTest):
+    @pytest.fixture
+    def integrations(self):
+        return [FIRST_INTEGRATION]
+
+    def test_catalog_parsing(self, catalogs, project):
+        with self.setup_test(catalogs, project):
+            run_dbt(["run"])
+            write_integration = project.adapter.get_catalog_integration("my_write_integration")
```

Comment:
This needs to be updated with the `catalog_name` updates.
```suggestion
            write_integration = project.adapter.get_catalog_integration("test_catalog")
```